src/main/java/org/ow2/proactive/catalog/service/CatalogObjectReportService.java

```diff
@@ -0,0 +1,104 @@
+import java.util.TreeSet;
+
+import org.ow2.proactive.catalog.dto.CatalogObjectMetadata;
+import org.ow2.proactive.catalog.report.CatalogObjectReportPDFGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import lombok.extern.log4j.Log4j2;
+
+
+/**
+ * @author ActiveEon Team
+ */
+@Log4j2
+@Service
+@Transactional
+public class CatalogObjectReportService {
+
+    @Autowired
+    private CatalogObjectService catalogObjectService;
+
+    @Autowired
+    private CatalogObjectReportPDFGenerator catalogObjectReportPDFGenerator;
+
+    public byte[] generateBytesReport(List<String> authorisedBucketsNames, Optional<String> kind,
+            Optional<String> contentType) {
+
+        if (authorisedBucketsNames.isEmpty()) {
+            return new byte[0];
+        }
+
+        List<CatalogObjectMetadata> metadataList = getListOfCatalogObjects(kind, contentType, authorisedBucketsNames);
+
+        TreeSet<CatalogObjectMetadata> orderedObjectsPerBucket = sortObjectsPerBucket(metadataList);
+
+        return catalogObjectReportPDFGenerator.generatePDF(orderedObjectsPerBucket);
+
+    }
+
+    private List<CatalogObjectMetadata> getListOfCatalogObjects(Optional<String> kind, Optional<String> contentType,
+            List<String> authorisedBucketsNames) {
+        List<CatalogObjectMetadata> metadataList;
+
+        if (kind.isPresent() && contentType.isPresent()) {
+            metadataList = catalogObjectService.listCatalogObjectsByKindAndContentType(authorisedBucketsNames,
+                                                                                       kind.get(),
+                                                                                       contentType.get());
+        } else if (contentType.isPresent()) {
+            metadataList = catalogObjectService.listCatalogObjectsByContentType(authorisedBucketsNames,
+                                                                                contentType.get());
+        } else if (kind.isPresent()) {
+            metadataList = catalogObjectService.listCatalogObjectsByKind(authorisedBucketsNames, kind.get());
+        } else {
+            metadataList = catalogObjectService.listCatalogObjects(authorisedBucketsNames);
+        }
+
+        return metadataList;
+    }
+
+    private TreeSet<CatalogObjectMetadata> sortObjectsPerBucket(List<CatalogObjectMetadata> metadataList) {
+
+        Comparator<CatalogObjectMetadata> sortBasedOnName = Comparator.comparing(CatalogObjectMetadata::getBucketName);
+        sortBasedOnName = sortBasedOnName.thenComparing(Comparator.comparing(CatalogObjectMetadata::getProjectName));
+        sortBasedOnName = sortBasedOnName.thenComparing(Comparator.comparing(CatalogObjectMetadata::getName));
+
+        TreeSet<CatalogObjectMetadata> sortedObjects = new TreeSet<CatalogObjectMetadata>(sortBasedOnName);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)